core/src/main/scala/akka/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -3,20 +3,34 @@
  * Copyright (C) 2019 - 2020 Lightbend Inc. <https://www.lightbend.com>
  */
 
-package akka.persistence.jdbc
-package journal.dao
+package akka.persistence.jdbc.journal.dao
 
-import akka.persistence.jdbc.config.JournalTableConfiguration
+import akka.persistence.jdbc.config.{ EventJournalTableConfiguration, EventTagTableConfiguration }
+import akka.persistence.jdbc.journal.dao.JournalTables.{ JournalAkkaSerializationRow, TagRow }
 import slick.jdbc.JdbcProfile
 
-class JournalQueries(val profile: JdbcProfile, override val journalTableCfg: JournalTableConfiguration)
+import scala.concurrent.ExecutionContext
+
+class JournalQueries(
+    val profile: JdbcProfile,
+    override val journalTableCfg: EventJournalTableConfiguration,
+    override val tagTableCfg: EventTagTableConfiguration)
     extends JournalTables {
+
   import profile.api._
 
-  private val JournalTableC = Compiled(JournalTable)
+  val insertAndReturn =
+    JournalTable.returning(JournalTable.map(_.ordering))
+  private val TagTableC = Compiled(TagTable)
 
-  def writeJournalRows(xs: Seq[JournalRow]) =
-    JournalTableC ++= xs.sortBy(_.sequenceNumber)
+  def writeJournalRows(xs: Seq[(JournalAkkaSerializationRow, Set[String])])(implicit ec: ExecutionContext) = {
+    val sorted = xs.sortBy((event => event._1.sequenceNumber))
+    for {
+      ids <- insertAndReturn ++= sorted.map(_._1)
+      tagInserts = ids.zip(sorted.map(_._2)).flatMap { case (id, tags) => tags.map(tag => TagRow(id, tag)) }
+      _ <- TagTableC ++= tagInserts
+    } yield ()
```

Comment:
here we could do: 

```scala
val (events, tags) = sorted.unzip
```

We would traverse it once instead of twice (2 maps) and we avoid using ._1 and ._2 (less readable).